server/controllers/recipe.js

```diff
@@ -0,0 +1,323 @@
+     */
+    query.limit = req.query.limit || 1;
+    query.offset = req.query.offset || 0;
+
+    /**
+     * query the db for all recipes
+     * left join Reviews as reviews
+     * left join Favourites as favourites
+     * left join left join Votings as votings
+     * ordered by 'id' descending
+     */
+    Recipe
+      .findAndCountAll({
+        order: [['id', 'DESC']],
+        limit: query.limit,
+        offset: query.offset,
+      })
+      .then((recipe) => {
+        if (recipe.rows.length <= 0) {
+          return res.status(404).send({ message: 'No recipe(s) was found!' });
+        }
+
+        /**
+         * pass query limit, query offset, recipedata.count to pagenate helper
+         * and return totalCount, currentPage, pageCount, and pageSize
+         * to pagenation
+         */
+        const pagenation = pagenate(query.limit, query.offset, recipe.count);
+
+        return res.status(200).send(
+          {
+            pagenation,
+            recipes: recipe.rows,
+          }
+        );
+      })
+      .catch(error => res.status(400).send({ message: error.message }));
+  },
+
+  /**
+    * @description retrieve recipe controller
+    * @param {Object} req - Request object
+    * @param {Object} res - Response object
+    * @returns {Object} json - payload
+    */
+  retrieveRecipe(req, res) {
+    Recipe
+      .findOne({
+        where: {
+          id: req.params.id,
+        },
+      })
+      .then((recipe) => {
+        if (recipe) {
+          recipe.increment('views');
+          return res.status(200).send(recipe);
+        }
+        return res.status(404).send({ message: 'Recipe not found' });
+      })
+      .catch(error => res.status(400).send({ message: error.message }));
+  },
+
+  /**
+     * @description update recipe controller
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  updateRecipe(req, res) {
+    // query db to check if recipe exist
+    Recipe
+      .findById(req.params.id)
+      .then((recipe) => {
+        if (!recipe) {
+          return res.status(404).send({ message: 'Recipe not found' });
+        }
+        Recipe
+          .find({
+            where: {
+              addedBy: req.decoded.user.id,
+              id: req.params.id,
+            },
+          })
+          .then((recipe) => {
+            if (!recipe) {
+              return res.status(401).send({
+                message: 'Operation not allowed. You can only update your recipe'
+              });
+            }
+            recipe
+              .update({
+                title: req.body.title || recipe.title,
+                description: req.body.description || recipe.description,
+                ingredients: req.body.ingredients || recipe.ingredients,
+                procedures: req.body.procedures || recipe.procedures,
+                imageUrl: req.body.imageUrl || recipe.imageUrl,
+              })
```

Comment:
You can do Object destructuring for a neater look here.
Ex.
`const sample = { name: 'Professor X', age: undefined };`
to obtain name and age; I can simply do:
` const { name, age } = sample;` 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment